server/src/main/java/org/elasticsearch/cluster/coordination/ClusterBootstrapService.java

```diff
@@ -114,6 +114,13 @@ void onFoundPeersUpdated() {
             logger.trace("nodesMatchingRequirements={}, unsatisfiedRequirements={}, bootstrapRequirements={}",
                 nodesMatchingRequirements, unsatisfiedRequirements, bootstrapRequirements);
 
+            if (nodesMatchingRequirements.contains(transportService.getLocalNode()) == false) {
```

Comment:
I think we can check this in the constructor and just assert it here. Also we use the word `cancelling` rather than `skipping` indicating that we won't retry.

Reply:
The local discoverynode object is not available in the constructor yet (i.e. the transport has not bound at this point yet)

Reply:
Blasted laziness. Ok.